File: solutions/lua/resistor-color-trio/1/resistor-color-trio.lua
local color_map = {
  black = 0,
  brown = 1,
  red = 2,
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9,
}
return {
  decode = function(c1, c2, c3)
    local val = (color_map[c1] * 10 + color_map[c2]) * (10 ^ color_map[c3])
    if val > 1000000000 then return (val / 1000000000), "gigaohms" end
    if val > 1000000 then return (val / 1000000), "megaohms" end
    if val > 1000 then return (val / 1000), "kiloohms" end
    return val, "ohms"
  end
}

File: solutions/clojure/resistor-color-trio/1/src/resistor_color_trio.clj
(ns resistor-color-trio)

(def ^:private color->value (zipmap ["black" "brown" "red" "orange" "yellow"
                                     "green" "blue" "violet" "grey" "white"]
                                    (range)))
(defn resistor-label [colors]
  (let [[c1 c2 c3] (map color->value colors)
        resistance (long (* (+ (* 10 c1) c2)
                            (Math/pow 10 c3)))]
    (loop [resistance resistance
           [unit & units] [" ohms" " kiloohms" " megaohms" " gigaohms"]]
      (if (< resistance 1000)
        (str resistance unit)
        (recur (quot resistance 1000) units)))))
